### DISCUSSION TITLE
How do I change the default PATH environment variable for bash? | 如何修改 bash 默认的 PATH 环境变量？

### DISCUSSION THREAD

**Discussion by DOSforever** [cmderdev/cmder#2981] [Q&A]
在 Settings -> Startup -> Environment 中即不设置额外的有关 PATH 的环境变量，启动 bash 时也会自动生成默认的 PATH 环境变量，其中包括了 Windows 本身的 PATH 环境变量，奇怪的是还有些路径是我系统上根本就没有的，不知道它是如何生成的。那么除了 Windows 本身的环境变量外，其它由 Cmder 或 ConEmu 生成的 PATH 环境变量该如何修改呢？

---

In Settings -> Startup -> Environment, even if you don't set any additional PATH environment variables, bash will automatically generate default PATH environment variables when you start bash, including Windows' own PATH environment variables, but strangely enough, there are some paths that don't exist on my system,  I don't know how it was generated. So how do I change the PATH environment variables generated by Cmder or ConEmu other than the Windows environment variables?

Translated with DeepL.com (free version)

**Comment by daxgames** [comment]
You can edit `$CMDER_ROOT/config/user_profile.*` and set whatever env variables you want.

**Reply by DOSforever** [reply]
我的意思是如果我不额外编写任何配置文件，那些自动生成的默认的 PATH 是如何产生的，能否修改这些默认设置？在 $CMDER_ROOT/config/user_profile.* 中编写的 path 是添加到默认的 PATH 环境变量中还是覆盖掉默认的 PATH 环境变量？

---

I mean if I don't write any additional profiles, how are those auto-generated default PATH generated and can I change those default settings? Is the path written in $CMDER_ROOT/config/user_profile.* added to the default PATH environment variable or does it ovewrite the default PATH environment variable?

**Reply by daxgames** [reply]
They come from the environment of the system, they're not generated by anything they are the environment that you ran Cmder from.  You would have to change the entire environment of the entire Windows system before launching Cmder. 

Your options are to do what I said or do what you said by setting the env within the Comemu env settings dialog.

**Reply by DOSforever** [reply]
但是有个问题我不明白：

假如不编写额外的 PATH 变量，bash 启动后的 PATH 变量由三部分组成（假定我的 Cmder 安装目录为 D:\Cmder）
>***/c/Users/winusername/bin***:/d/cmder/bin:/d/cmder/vendor/bin:/d/cmder/vendor/git-for-windows/bin:***/usr/local/bin***:/usr/bin:/bin:***/opt/bin***
:/C/Windows/system32:/C/Windows:/C/Windows/System32/Wbem
:/d/cmder:/usr/bin/vendor_perl:/usr/bin/core_perl

第一和第三部分是 bash 启动后固有的，中间第二部分是复制 Windows CMD 的 PATH 环境变量。第一部分中有三个路径是在我系统上没有的
***/c/Users/winusername/bin***
有 /c/Users/winusername/ 目录，但却没有其下的 *bin* 子目录
***/usr/local/bin***
有 /usr/local/ 目录，但却没有其下的 *bin* 子目录
***/opt/bin***
有 /opt/ 目录，但却没有其下的 *bin* 子目录

我想问下这些不存在的目录路径是如何产生的？能否对第一和第三部分的路径进行编辑？

---

But there's one thing I don't understand:

If you don't write additional PATH variables, the PATH variable after bash starts consists of three parts (assuming my Cmder installation directory is D:\Cmder)
>***/c/Users/winusername/bin***:/d/cmder/bin:/d/cmder/vendor/bin:/d/cmder/vendor/git-for-windows/bin:***/usr/local/bin***:/usr/bin:/bin:***/opt/bin***
:/C/Windows/system32:/C/Windows:/C/Windows/System32/Wbem
:/d/cmder:/usr/bin/vendor_perl:/usr/bin/core_perl

The first and third parts are inherent to the bash startup, and the second part in the middle is copying the Windows CMD PATH environment variable. There are three paths in the first part that are not present on my system
***/c/Users/winusername/bin***
There is the /c/Users/winusername/ directory, but no *bin* subdirectory under it.
***/usr/local/bin***
There is a /usr/local/ directory, but no *bin* subdirectory under it.
***/opt/bin***
There is a directory /opt/, but no *bin* subdirectory under it.

I would like to ask how these non-existing directory paths are created? Is it possible to edit the first and third part of the PATH?
